daemon/mgr/container.go

```diff
@@ -1901,3 +2039,48 @@ func (mgr *ContainerManager) execProcessGC() {
 		}
 	}
 }
+
+// hasUserDefinedIPAddress returns whether the passed endpoint configuration contains IP address configuration
+func hasUserDefinedIPAddress(epConfig *types.EndpointSettings) bool {
```

Comment:
This kind of function should be encapsulated in network utils or something like that. Here in container manager is quite unreasonable.